views/js/scoring/processor/expressions/operators/mathOperator.js

```diff
@@ -0,0 +1,261 @@
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technlogies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+/**
+ * The mathOperator processor.
+ * @see http://www.imsglobal.org/question/qtiv2p1/imsqti_infov2p1.html#element106451
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+define([
+    'lodash'
+], function(_){
+    'use strict';
+
+    /**
+     * Process operands and returns the mathOperator result.
+     * @type {OperandProcessor}
+     * @exports taoQtiItem/scoring/processor/expressions/operators/mathOperator
+     */
+    var mathOperatorProcessor = {
+
+        accuracy : 0.000000000000001,
+
+        constraints : {
+            minOperand : 1,
+            maxOperand : -1,
+            cardinality : ['single'],
+            baseType : ['float','integer']
+        },
+
+        operands   : [],
+
+        /**
+         * Process the mathOperator of the operands.
+         * @returns {?ProcessingValue} the and or null
+         */
+        process : function(){
+
+            var result = {
+                cardinality : 'single',
+                baseType : 'float'
+            };
+
+            //if at least one operand is null, then break and return null
+            if(_.some(this.operands, _.isNull) === true){
+                return null;
+            }
+
+            var name = this.expression.attributes.name;
+
+            if (!_.contains(Object.keys(functions), name)) {
+                return null;
+            }
+
+            var operands = this.preProcessor.parseOperands(this.operands).value();
+
+            if (_.some(operands, _.isNaN)) {
+                return null;
+            }
+
+            if (_.isFunction(functions[name])) {
+                result.value = functions[name](operands[0]);
+            } else {
+                result.value = functions[name].func(operands);
+            }
+
+            if (_.isNull(result.value) || _.isNaN(result.value)) {
+                return null;
+            }
+            return result;
+        }
+    };
+
+    var functions = {
+        sin: Math.sin,
+        cos: Math.cos,
+        tan: Math.tan,
+        sec: reciprocalFactory(Math.cos),
+        csc: reciprocalFactory(Math.sin),
+        cot: reciprocalFactory(Math.tan),
+        asin: Math.asin,
+        acos: Math.acos,
+        atan: Math.atan,
+        atan2: {
```

Comment:
I realized that is not very transparent naming was used there, it is for splitting non standard cases, when function require more than one argument ( like `atan2` ), originally  there are was extra property with number of required arguments